File: repeater.py
import groups
import send

global last
global times
dd = "打断"
dg = "出现了打断怪"
last = ""
times = 1


async def handle(msg: str):
    global last
    global times
    if(msg == last):
        times += 1
        if(times == 4):
            times = 1
            if(last == dd):
                await send.major(dg)
                last = dg
            else:
                await send.major(dd)
                last = dd
    else:
        times = 1
        last = msg

File: LemonApp/Repeater.cs
namespace Clansty.tianlang
{
    static class Repeater
    {
        private static string last;
        private static int i;

        private const string dd = "打断";
        private const string dg = "出现了打断怪";

        public static void Enter(string msg)
        {
            if (last == msg)
            {
                i++;
                if (i == 4)
                {
                    if (msg == dd)
                    {
                        S.Major(dg);
                        last = dg;
                    }
                    else
                    {
                        S.Major(dd);
                        last = dd;
                    }
                    i = 1;
                }
            }
            else
            {
                last = msg;
                i = 1;
            }
        }
    }
}
